pkg/generate/code/check.go

```diff
@@ -132,37 +138,76 @@ func checkRequiredFieldsMissingFromShape(
 			missing = append(missing, primaryARNCondition)
 			continue
 		}
-		cleanMemberNames := names.New(memberName)
-		cleanMemberName := cleanMemberNames.Camel
 
-		resVarPath := koVarName
-		// Check that the field has potentially been renamed
-		renamedName, wasRenamed := r.InputFieldRename(
-			op.Name, memberName,
-		)
-		_, found := r.SpecFields[renamedName]
-		if found && !wasRenamed {
-			resVarPath = resVarPath + r.Config().PrefixConfig.SpecField + "." + cleanMemberName
-		} else if found {
-			resVarPath = resVarPath + r.Config().PrefixConfig.SpecField + "." + renamedName
-		} else {
-			_, found = r.StatusFields[memberName]
-			if !found {
+		resVarPath, err := getSanitizedMemberPath(memberName, r, op, koVarName)
+		if err != nil {
+			// If it isn't in our spec/status fields, we have a problem!
+			msg := fmt.Sprintf(
+				"GENERATION FAILURE! there's a required field %s in "+
+					"Shape %s that isn't in either the CR's Spec or "+
+					"Status structs!",
+				memberName, shape.ShapeName,
+			)
+			panic(msg)
+		}
+		missing = append(missing, fmt.Sprintf("%s == nil", resVarPath))
+	}
+
+	// Loop over the not-required member fields in the shape and determine
+	// if an ID field can be populated by an ID in either the Status or the Spec of the resource
+	// and generate an if condition checking for all required fields having non-nil
+	// corresponding resource Spec/Status values (required for ReadMany)
+	for field := range notRequiredFields {
+		crKind := strings.ToLower(r.Kind)
+		if strings.Contains(strings.ToLower(field), crKind) {
+			pluralize := pluralize.NewClient()
+			// CR shouldn't have plural of its kind as a field
+			singular := pluralize.Singular(field)
+
+			resVarPath, err := getSanitizedMemberPath(singular, r, op, koVarName)
+			if err != nil {
 				// If it isn't in our spec/status fields, we have a problem!
 				msg := fmt.Sprintf(
 					"GENERATION FAILURE! there's a required field %s in "+
 						"Shape %s that isn't in either the CR's Spec or "+
 						"Status structs!",
-					memberName, shape.ShapeName,
+					singular, shape.ShapeName,
 				)
 				panic(msg)
 			}
-			resVarPath = resVarPath + r.Config().PrefixConfig.StatusField + "." + cleanMemberName
+			missing = append(missing, fmt.Sprintf("%s == nil", resVarPath))
+			// Not expecting multiple ID fields
+			break
 		}
-		missing = append(missing, fmt.Sprintf("%s == nil", resVarPath))
 	}
+
 	// Use '||' because if any of the required fields are missing the object
 	// is not created yet
 	missingCondition := strings.Join(missing, " || ")
 	return fmt.Sprintf("%sreturn %s\n", indent, missingCondition)
 }
+
+// getSanitizedMemberPath takes a shape member field, checks for renames, checks for existence in Spec and Status, then
+// constructs and returns the var path. Returns error if memberName is not present in either Spec or Status.
+func getSanitizedMemberPath(memberName string, r *model.CRD, op *awssdkmodel.Operation, koVarName string) (string, error) {
```

Comment:
Nice little cleanup function, thank you. Please remember to wrap comments at 80 chars and also for long (>80 char) function signatures like this, I like to break the function parameters by newline:

```go
func getSanitizedMemberPath(
	memberName string,
	r *model.CRD,
	op *awssdkmodel.Operation,
	koVarName string,
) (string, error) {
...
}
```